### PR TITLE
[dynamo 3.11] fix 3.11.2 issues

### PR DESCRIPTION
**Pull request by williamwen42**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #96511
* #98104
* __->__ #98364

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @soumith @anijain2305 @Xia-Weiwen @desertfire

### CODE REVIEW

File: torch/csrc/dynamo/eval_frame.c [hunk 1]
```diff
@@ -592,6 +592,12 @@ inline static PyObject* eval_custom_code(
   THP_EVAL_API_FRAME_OBJECT* shadow = shadow_obj->f_frame;
   Py_XINCREF(frame->f_func->func_closure);
   shadow->f_func->func_closure = frame->f_func->func_closure;
+  // NOTE: in Python 3.11.1+, PyFrame_New changes prev_instr, causing
```
Comment (albanD) [inline torch/csrc/dynamo/eval_frame.c:595]: I'm not sure what this comment means? Dont we want the frame to be properly valid, including prev_instr ?
Why are we resetting the value to something invalid?

Comment (williamwen42) [reply torch/csrc/dynamo/eval_frame.c:595]: It doesn't seem like CPython uses PyFrame_New for regular frame evaluation. PyFrame_New marks the frame as "complete" (by setting prev_instr to the first RESUME instruction), but frames for frame evaluation expect prev_instr to be the instruction before the first instruction. There may be more underlying issues with using PyFrame_New, but I'll address them in future PRs.

Comment (albanD) [reply torch/csrc/dynamo/eval_frame.c:595]: Ok sounds good to unblock but most we should most likely clean it up.

**Review by albanD — APPROVED** [review]
SGTM

**Pull request closed without merging**